### ISSUE TITLE
Snippet and Image links Broken ?

### ISSUE THREAD

**Issue by MrAsrani1312** [freeCodeCamp/guides#233]
So, I have written the intro guide under C++ in guides.
The code snippet does not show the last 2 lines 
```C++
<endl;
return 0;
}
```
and the image link doesn't show the image but shows the title I have given it.
All other image links are not working 😭 
Can anyone help me out to solve this issue?
The Main website : 
![image](https://user-images.githubusercontent.com/24650979/29996048-e1f152a6-9014-11e7-85bd-e0efc26b250a.png)

The GitHub output for the article I've written :
![image](https://user-images.githubusercontent.com/24650979/29996012-362d4af6-9014-11e7-9430-010e897a73cd.png)
 The code for the same : 
![image](https://user-images.githubusercontent.com/24650979/29996029-751ef00c-9014-11e7-8147-84cb58f9c67a.png)

Help needed ASAP so that all other articles written by me do not have this issue .
Thanks.
Hitarth.

**Comment by Bouncey** [comment]
This is the same issue as with #239 the markdown-to-html plugin we are using with Gatsby is converting things to html when it really shouldn't. In this case `<iostream>` is converted into an html element, which breaks everything else coming after it.

**Comment by Bouncey** [comment]
It's not a bug with the article or how we do things, its an issue with `gatsby-transformer-remark`

**Comment by Bouncey** [comment]
Issue raised at wooorm/remark#297

**Comment by Bouncey** [comment]
@systimotic I'm not deving at the minute so I can't log anything out to see, but my hunch is the issue is in here
 ```js
   async function getHTML(markdownNode) {
      const cachedHTML = await cache.get(htmlCacheKey(markdownNode))
      if (cachedHTML) {
        return cachedHTML
      } else {
        const html = await new Promise((resolve, reject) => {
          getAST(markdownNode).then(ast => {
            resolve(
              hastToHTML(toHAST(ast, { allowDangerousHTML: true }), {
                allowDangerousHTML: true,
              })
            )
          })
        })

        // Save new HTML to cache and return
        cache.set(htmlCacheKey(markdownNode), html)
        return html
      }
    }
```
both `hastToHTML` and `toHAST` are from the same guy that made `remark`

**Comment by systimotic** [comment]
That's what I was thinking as well, though I haven't quite figured it out yet. It looks like it may also be related to `gatsby-remark-prismjs`.

**Comment by Bouncey** [comment]
Seems to be fixed in the latest update

**Comment by systimotic** [comment]
@Bouncey This doesn't seem fixed in the latest update for me. Even after removing all node_modules and reinstalling all the packages (on the #281 branch), this problem persists.

**Comment by Bouncey** [comment]
hmmmmm.............. 

I have updated the PR comment so it will not close this if merged

**Comment by systimotic** [comment]
@Bouncey I found some more examples that show the same issue, only confirming our suspicions https://guide.freecodecamp.org/javascript/standard-objects/array/array-prototype-reverse and most of the other array articles. Some of the string articles have this issue as well. Again, because of an HTML element inside of the code block. 
Interestingly, the HTML blocks are not affected https://guide.freecodecamp.org/bootstrap/bootstrap-navigation-bar

I finally tested this with `remark` disabled, and the issue did go away, although the links in the code blocks look really weird still. Should we disable it as a temporary fix and report it as an issue over at Gatsby?
